plutus-contract/test/Spec/TxConstraints/MustSpendPubKeyOutput.hs

```diff
@@ -69,10 +69,13 @@ baseLovelaceLockedByScript :: Integer
 baseLovelaceLockedByScript = lovelacePerInitialUtxo `div` 2
 
 mustSpendPubKeyOutputContract :: [TxOutRef] -> [TxOutRef] -> Ledger.PaymentPubKeyHash -> Contract () Empty ContractError ()
-mustSpendPubKeyOutputContract offChainTxOutRefs onChainTxOutRefs pkh = do
+mustSpendPubKeyOutputContract = mustSpendPubKeyOutputContract' []
+
+mustSpendPubKeyOutputContract' :: [Ledger.PaymentPubKeyHash] -> [TxOutRef] -> [TxOutRef] -> Ledger.PaymentPubKeyHash -> Contract () Empty ContractError ()
+mustSpendPubKeyOutputContract' keys offChainTxOutRefs onChainTxOutRefs pkh = do
     let lookups1 = Constraints.typedValidatorLookups typedValidator
-        tx1 = Constraints.mustPayToTheScriptWithDatumInTx onChainTxOutRefs
-            $ Ada.lovelaceValueOf baseLovelaceLockedByScript
+        tx1 = Constraints.mustPayToTheScriptWithDatumInTx onChainTxOutRefs (Ada.lovelaceValueOf baseLovelaceLockedByScript)
+            <> foldMap Constraints.mustBeSignedBy keys
```

Comment:
Why is this now needed?

Reply:
That's possible. Not to worry. Thanks!

Reply:
Actually, something to note is that `mustBeSignedBy` includes required signer in the txbody, which is redundant in the case where onchain validation for this witness is not performed.